### PR TITLE
SYNCT-185: Added a whitelist to the SYNC2 configuration

### PR DESCRIPTION
**Pull request by wojtekbu**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/sync2/api/model/configuration/SyncConfiguration.java [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 public class SyncConfiguration {
 
     private GeneralConfiguration general;
     private SyncMethodConfiguration push;
     private SyncMethodConfiguration pull;
+    private WhitelistConfiguration whitelist;
 
-    public SyncConfiguration() { }
+    public SyncConfiguration() {
+        whitelist = getDefaultWhitelistConfiguration();
```
Comment (areklalo) [inline api/src/main/java/org/openmrs/module/sync2/api/model/configuration/SyncConfiguration.java]: Should we do the same with other fields?

File: api/src/main/java/org/openmrs/module/sync2/api/model/configuration/WhitelistConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.List;
+import java.util.Objects;
+
+public class WhitelistConfiguration {
+
+    private boolean enabled;
+    private List<String> instanceIds;
+
+    public WhitelistConfiguration() {}
```
Comment (areklalo) [inline api/src/main/java/org/openmrs/module/sync2/api/model/configuration/WhitelistConfiguration.java]: What do you think about Initialization the instanceIds variable with empty ArrayList?

File: api/src/main/java/org/openmrs/module/sync2/api/model/configuration/SyncConfiguration.java (hunk 1 above)
Comment (wojtekbu) [reply api/src/main/java/org/openmrs/module/sync2/api/model/configuration/SyncConfiguration.java]: The whitelist is a functionality which isn't obligatory. That's why a user does not have to provide configuration for it. The rest of configurations support core functionalities, so they should be configurated.

File: api/src/main/java/org/openmrs/module/sync2/api/model/configuration/WhitelistConfiguration.java (hunk 2 above)
Comment (wojtekbu) [reply api/src/main/java/org/openmrs/module/sync2/api/model/configuration/WhitelistConfiguration.java]: OK

File: api/src/main/java/org/openmrs/module/sync2/api/model/configuration/SyncConfiguration.java (hunk 1 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/sync2/api/model/configuration/SyncConfiguration.java]: Okay, that makes sense.

**Pull request merged**